layer/nrlf/nrlf/core/common_search_steps.py

```diff
@@ -0,0 +1,92 @@
+from enum import Enum
+from logging import Logger
+from typing import Any, Dict, Optional, Union
+
+from lambda_pipeline.types import FrozenDict, PipelineData
+from pydantic import Field
+from typing_extensions import Annotated
+
+from nrlf.consumer.fhir.r4.model import NextPageToken, RequestQuerySubject
+from nrlf.core.common_steps import make_common_log_action
+from nrlf.core.constants import DbPrefix
+from nrlf.core.errors import assert_no_extra_params
+from nrlf.core.model import (
+    ConsumerRequestParams,
+    CountRequestParams,
+    PaginatedResponse,
+    key,
+)
+from nrlf.core.repository import (
+    PAGE_ITEM_LIMIT,
+    Repository,
+    custodian_filter,
+    type_filter,
+)
+from nrlf.core.validators import validate_type_system
+
+log_action = make_common_log_action()
+
+
+class LogReference(Enum):
+    COMMONSEARCH001 = "Searching for document references"
+
+
+@log_action(log_reference=LogReference.COMMONSEARCH001)
+def get_paginated_document_references(
```

Comment:
possible to split this function into a few parts? would be better for logging and debugging